### PR TITLE
Fix: handle the case where env var is empty

### PR DESCRIPTION
**Pull request by xingyaoww**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

Fix the issue @mamoodi reported in slack: https://opendevin.slack.com/archives/C06TZ1VCCAD/p1721328419082809?thread_ts=1721247898.211409&cid=C06TZ1VCCAD

```
14:46:37 - opendevin:ERROR: session.py:113 - Error creating controller: cannot access local variable 'key' where it is not associated with a value
Traceback (most recent call last):
  File "/Users/mahmoudwork/Desktop/OpenDevin_Forks/tobifork/OpenDevin/opendevin/server/session/session.py", line 105, in _initialize_agent
    await self.agent_session.start(
  File "/Users/mahmoudwork/Desktop/OpenDevin_Forks/tobifork/OpenDevin/opendevin/server/session/agent.py", line 56, in start
    await self._create_runtime(runtime_name, sandbox_config)
  File "/Users/mahmoudwork/Desktop/OpenDevin_Forks/tobifork/OpenDevin/opendevin/server/session/agent.py", line 83, in _create_runtime
    await self.runtime.ainit()
  File "/Users/mahmoudwork/Desktop/OpenDevin_Forks/tobifork/OpenDevin/opendevin/runtime/server/runtime.py", line 84, in ainit
    await super().ainit()
  File "/Users/mahmoudwork/Desktop/OpenDevin_Forks/tobifork/OpenDevin/opendevin/runtime/runtime.py", line 90, in ainit
    await self.add_env_var(self.DEFAULT_ENV_VARS)
  File "/Users/mahmoudwork/Desktop/OpenDevin_Forks/tobifork/OpenDevin/opendevin/runtime/runtime.py", line 140, in add_env_var
    raise RuntimeError(f'Failed to add {key} to environment: {obs}')
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnboundLocalError: cannot access local variable 'key' where it is not associated with a value
```
---

**Give a summary of what the PR does, explaining any non-trivial design decisions**

Fix the potential issue and add a test case to test setting empty env var.

---
**Other references**

### CODE REVIEW

**Review by tobitege — APPROVED** [review]
LGTM, thanks

File: opendevin/runtime/runtime.py [hunk 1]
```diff
@@ -115,16 +115,20 @@ def init_runtime_tools(
 
     # ====================================================================
 
-    async def add_env_var(self, vars):
+    async def add_env_var(self, env_vars: dict[str, str]) -> None:
```
[the remark this answers is not in the document]
Comment (xingyaoww) [reply opendevin/runtime/runtime.py]: done!
